tensorflow/lite/micro/memory_planner/non_persistent_buffer_planner_shim.cc

```diff
@@ -20,16 +20,21 @@ limitations under the License.
 namespace tflite {
 
 NonPersistentMemoryPlannerShim::NonPersistentMemoryPlannerShim(
-    const BufferPlan* offline_buffer_plan)
-    : offline_buffer_plan_(offline_buffer_plan) {}
+    const BufferPlan* buffer_plan)
+    : buffer_plan_(buffer_plan), buffer_request_count_(0) {}
 
 NonPersistentMemoryPlannerShim::~NonPersistentMemoryPlannerShim() {}
 
 TfLiteStatus NonPersistentMemoryPlannerShim::AddBuffer(
     tflite::ErrorReporter* error_reporter, int size, int first_time_used,
     int last_time_used) {
-  MicroPrintf("Unsupported operation");
-  return kTfLiteError;
+  buffer_request_count_++;
+  if (buffer_request_count_ > buffer_plan_->buffer_count) {
+    MicroPrintf("Buffer request %d exceeds the range %d", buffer_request_count_,
```

Comment:
I think this message could be clearer. Something like "Attempting to add %d buffers, only %d allowed in given buffer plan." As a user, I would find it easier to trace this back and increase the number of buffers in my plan.

Same comment applies to the MicroPrintf below.